### ISSUE TITLE
Duplicate function name constants in trace IR.

### ISSUE THREAD

**Issue by ptersilie** [ykjit/yk#664]
The trace IR currently contains hundreds of duplicate string constants.
```
@1 = private unnamed_addr constant [7 x i8] c"interp\00", align 1
@2 = private unnamed_addr constant [7 x i8] c"interp\00", align 1
@3 = private unnamed_addr constant [7 x i8] c"interp\00", align 1
...
```
This is because every guard failure needs to generate an array with frame information including the function name of each stack frame. These strings are not reused, so we get a new constant for each function on the stack for each guard failure. This quickly adds up.

This is related to https://github.com/ykjit/yk/issues/646.

**Comment by ltratt** [comment]
Good point: this is something that `trace_chewer` can make us ignore, because it just knocks all that stuff out of the output!

**Comment by ptersilie** [comment]
No not yet. It's on my todo list. It has to do with the frame information we pass into guard failures. Should be fairly easy to cache.
